From a7b55392abdd3a85bbe1c621c1d84a937cfff14e Mon Sep 17 00:00:00 2001
From: Lin Jianhua <linjh@rock-chips.com>
Date: Wed, 16 Sep 2020 16:36:00 +0800
Subject: [PATCH] HACK: media: videobuf2: allow cache hints on all memory types

This is a workaround to bypass RKISP dma buf sync between MIPI TX(mmap)
and MIPI RX(userptr|dmabuf).

Change-Id: I3d4966f35f2f3330816ac8a3d614486aa3276ebf
Signed-off-by: Lin Jianhua <linjh@rock-chips.com>
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 include/media/videobuf2-core.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/media/videobuf2-core.h b/include/media/videobuf2-core.h
index 0a99d8742780..77350c91616c 100644
--- a/include/media/videobuf2-core.h
+++ b/include/media/videobuf2-core.h
@@ -591,6 +591,9 @@ struct vb2_queue {
  */
 static inline bool vb2_queue_allows_cache_hints(struct vb2_queue *q)
 {
+#ifdef CONFIG_ARCH_ROCKCHIP
+	return q->allow_cache_hints;
+#endif
 	return q->allow_cache_hints && q->memory == VB2_MEMORY_MMAP;
 }
 
-- 
2.35.3

